clients/stellarcore/client.go

```diff
@@ -332,3 +289,100 @@ func (c *Client) simpleGet(
 
 	return req, nil
 }
+
+// rawPost returns a new POST request to the connected stellar-core using the
+// provided path and the params values encoded as the request body to construct
+// the result.
+func (c *Client) rawPost(
+	ctx context.Context,
+	newPath string,
+	params string,
+) (*http.Request, error) {
+	u, err := url.Parse(c.URL)
+	if err != nil {
+		return nil, errors.Wrap(err, "unparseable url")
+	}
+
+	u.Path = path.Join(u.Path, newPath)
+	newURL := u.String()
+
+	var req *http.Request
+	req, err = http.NewRequestWithContext(
+		ctx,
+		http.MethodPost,
+		newURL,
+		bytes.NewBuffer([]byte(params)))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create request")
+	}
+
+	return req, nil
+}
+
+// makeLedgerKeyRequest is a generic method to perform a request in the form
+// `key=...&key=...&ledgerSeq=...` which is useful because several Stellar Core
+// endpoints all use this request format. Be sure to pass `target` by reference.
+func (c *Client) makeLedgerKeyRequest(
+	ctx context.Context,
+	target interface{},
+	endpoint string,
+	ledgerSeq uint32,
+	keys ...xdr.LedgerKey,
+) error {
+	q, err := buildMultiKeyRequest(keys...)
+	if err != nil {
+		return err
+	} else if ledgerSeq >= 2 { // optional param
```

Comment:
nit: no need for else if since we return in the if statement above
```suggestion
	if ledgerSeq >= 2 { // optional param
```